solr/solrj/src/java/org/apache/solr/client/solrj/impl/LBHttp2SolrClient.java

```diff
@@ -257,6 +265,24 @@ private Cancellable doRequest(
             req.getRequest(),
             null,
             new AsyncListener<>() {
+              private final long requestSubmitTimeNanos = System.nanoTime();
```

Comment:
Just curious: Any particular reasons that we can using nanoTime over currentTimeMillis here? :) Is it more resistant to time drift or some other reasons?

Reply:
Yes! `System.currentTimeMillis()` is absolute wrt "real" wallclock time and must do extra work to adjust/correct accordingly. When all you need is relative timing, nanotime makes no such guarantees (it can even be negative!), so is more efficient and more appropriate for the kind of thing we're doing here, or micro-benchmarking, etc.